### PR TITLE
Implementing folds for new tree-sitter

### PR DESCRIPTION
**Pull request by mauricioszabo**
Just like it says on the title.

Tree-sitter, originally, have no implementation of folding. I basically copied this code from the NeoVIM implementation, which seems to be the most advanced on this implementation.

### CODE REVIEW

File: src/grammar-registry.js [hunk 1]
```diff
@@ -227,18 +233,15 @@ module.exports = class GrammarRegistry {
...
-    }
+    this.forEachGrammar(grammar => {
+      const score = this.getGrammarScore(grammar, filePath, fileContents);
+      if (score > highestScore || bestMatch == null) {
+        bestMatch = grammar;
+        highestScore = score;
+      }
+    });
+    return { grammar: bestMatch, score: highestScore };
+    // }
```
Comment (confused-Techie) [inline src/grammar-registry.js:244]:
```suggestion
```

File: src/wasm-tree-sitter-language-mode.js [hunk 2]
```diff
@@ -103,43 +112,45 @@ class WASMTreeSitterLanguageMode {
...
-      names.forEach(name => {
-        if(!this.scopeNames.get(name)) {
-          this.lastId += 2
-          const newId = this.lastId;
-          this.scopeNames.set(name, newId)
-          this.scopeIds.set(newId, name)
-        }
-      })
+    syntax.forEach(({node, name}) => {
+      // const node = capture.node
```
Comment (confused-Techie) [inline src/wasm-tree-sitter-language-mode.js:116]: Do we want to keep these comments? Or should we be fine to remove them?

**Review by confused-Techie — COMMENTED** [review]
So I won't pretend to fully understand every aspect of the code being used here.

But overall the code looks solid from a conventions standpoint, and am trusting you're testing it well.

Huge and appreciated effort here, just added a few comments to clean up in some small places.

---

Otherwise the tests:

* Editor tests on macOS is receiving 5 failures while we expect 0
* find-and-replace: 57 Failures, within the expected range
* symbols-view: 4 failures, but we expect only 2 failures (unsure why those would be failing here tbh)
* tree-view: 2 Failures, within the expected range
* Codacy Static Code Analysis has failed, but it seems it failed to properly analyze the commit, rather than any issues with it. So this can be ignored.

---

Otherwise this totally gets a lite approval from me, even though I'd like to read more and understand better what everything is doing here. But short of that happening I'm approving this getting merged.

**Comment by mauricioszabo** [comment]
@confused-Techie `symbols-view` might be a legitimate failure, I'll look at it

**Comment by mauricioszabo** [comment]
I'll address the test failures on branch `feature/modernize-tree-sitter` :)

**Pull request merged**